plumbing kmers to functions ; adding spaced seeds to composer

zeeev opened the pull request:
Please please check that the parameters are appropriately plumbed. The inception in composer is very deep.

Commit:
plumbing kmers to functions ; adding spaced seeds to composer

pb-cdunn commented on src/nibpkg/compose.nim:
```diff
@@ -18,11 +18,14 @@ proc retrieve_flanking_sequences_from_fai*(fastaIdx: Fai, chrom: string,
 
-proc kmerize(s: string, k: int = 25): seq[seed_t] =
-  return Dna(s).dna_to_kmers(k).seeds
+proc kmerize(s: string, k: int = 25, spaced_seeds : bool = false, space: int = 50): seq[seed_t] =
```
Simpler is to let `space == 0` be the default, and when you pass `50` then use spaced seeds.

But I might have a slightly better idea. If so, I'll push something in a few minutes ...

pb-cdunn replied:
Ok, I've pushed onto your branch, for that minor change plus actually using spaced-seeds for the initial "alt seqs" too. That might fix the problem. (I don't have time to run locally today.) Take a look.

Commit:
Make tests work on Mac w/ macports

It worked with homebrew because homebrew puts htslib
(and other libs) into the standard directory. Many people
eschew that practice, and I was advised to switch to macport.
But because MacOS strips DYLD_LIBRARY_PATH from env for any
sub-shell, I have to use install_tool to add an rpath.

Static linking would be another option. I can think of a few others
too. But this one works well enough.

Commit:
vim modeline

Commit:
Simplify new plumbing

Commit:
Use spaced seeds for initial alt seq too

Commit:
more var threading for spaced seeds
